src/main.cpp

```diff
@@ -0,0 +1,95 @@
+#include <iostream>
+#include <fstream>
+
+#include "DiffusionTensor.hpp"
+#include "FisherKolmogorov.hpp"
+
+#define DIMENSION 3
+
+int main(int argc, char *argv[]) {
+    Utilities::MPI::MPI_InitFinalize mpi_init(argc, argv);
+
+    double dext{0.05}, daxn{0.30}, T{1}, deltaT{0.1}, alpha{2};
+    int r = 1;
+    Point<DIMENSION> mass_center = {0.0, 0.0, 0.25};
+    Point<2> origin = {0,-0.4};
+    Point<DIMENSION> radial_center = {0, 5, 0};
+
+    std::string line,
+                w,
+                diffusion="CYLINDRICAL",
+                mesh_file_name="../mesh/brain-3D-scaled.msh";
+    std::ifstream f("./config.txt");
+    if (f.good()) {
+        while (std::getline(f, line)) {
+            std::istringstream iss(line);
+            iss >> w;
+            if (w == "r") {
+                iss >> r;
+            }  else if (w == "mesh_file"){
+                // char c;
+                // iss >> c;
+                std::string s(std::istreambuf_iterator<char>(iss), {});
+                mesh_file_name=s;
+                mesh_file_name.erase(0,1);
+            }else if (w == "dext") {
+                iss >> dext;
+            } else if (w == "daxn") {
+                iss >> daxn;
+            } else if (w == "T") {
+                iss >> T;
+            } else if (w == "dt") {
+                iss >> deltaT;
+            } else if (w == "alpha") {
+                iss >> alpha;
+            } else if (w == "diffusion") {
+                iss >> diffusion;
+                if (diffusion != "RADIAL" && diffusion != "CYLINDRICAL") {
+                    std::cerr << "Invalid diffusion tensor type\n";
+                    exit(1);  
+                }
+            } else if (w == "origin") {
+                iss  >> origin[0] >> origin[1];
+            } else if (w == "mass-center") {
+                for(int i = 0; i  < DIMENSION;i++) iss >> mass_center[i];
+            } else if (w == "radial-center") {
+                for(int i = 0; i  < DIMENSION;i++) iss >> mass_center[i];
+            } else if (w.length()==0 || w[0]=='#') {
+                // ignore empty lines and comments
+            } 
+            else {
+                std::cerr << "Invalid parameter: "<<w<<"\n";
+                exit(1);
+            }
+            
+        }
+    }
```

Comment:
You should consider looking into `dealii`'s `ParameterHandler` class for this sort of thing - it provides a much more streamlined and user-friendly interface.